api/occupancy/OccupancyDao.php

```diff
@@ -15,42 +15,44 @@ public static function processFeedback($feedback, $epochFeedback)
     {
         date_default_timezone_set('Europe/Brussels');
 
-        $stops = self::getVehicleStopsInfo($feedback["vehicle"]);
+        $stops = self::getVehicleStopsInfo($feedback['vehicle']);
         $errorCheck = 0;
-        $lastStation = "";
-
-        foreach ($stops->stop as $stop) {
-            if ($errorCheck > 0) {
-                if ($epochFeedback < intval($stop->time) + intval($stop["delay"])) {
-                    $feedback["from"] = $lastStation;
-                    self::processFeedbackOneConnection($feedback);
-                    break;
-                } else {
-                    $lastStation = $stop->station["URI"][0];
+        $lastStation = '';
+
+        if (!is_null($feedback['to'])) {
+            foreach ($stops->stop as $stop) {
+                if ($errorCheck > 0) {
+                    if ($epochFeedback < intval($stop->time) + intval($stop['delay'])) {
+                        $feedback['from'] = $lastStation;
+                        self::processFeedbackOneConnection($feedback);
+                        break;
+                    } else {
+                        $lastStation = $stop->station['URI'][0];
+                    }
                 }
-            }
 
-            if ($stop->station["URI"] == $feedback["from"] || $stop->station["URI"] == $feedback["to"]) {
-                if ($errorCheck == 0) {
-                    $lastStation = $stop->station["URI"][0];
-                }
+                if ($stop->station['URI'] == $feedback['from'] || $stop->station['URI'] == $feedback['to']) {
+                    if ($errorCheck == 0) {
+                        $lastStation = $stop->station['URI'][0];
+                    }
 
-                $errorCheck += 1;
-            }
+                    $errorCheck += 1;
+                }
 
-            if ($errorCheck == 2) {
-                self::processFeedbackOneConnection($feedback);
-                break;
+                if ($errorCheck == 2) {
+                    self::processFeedbackOneConnection($feedback);
+                    break;
+                }
             }
+        } else {
+            self::processFeedbackOneConnection($feedback);
         }
     }
 
-    private static function getVehicleStopsInfo($vehicle)
+    private static function getVehicleStopsInfo($URI)
     {
-        $url = "http://api.irail.be/vehicle/?id=BE.NMBS." . $vehicle;
-
         $curl = curl_init();
-        curl_setopt($curl, CURLOPT_URL, $url);
+        curl_setopt($curl, CURLOPT_URL, $URI);
```

Comment:
which URI is this? Can you give it a better name? it's like naming a variable "variable" or "identifier"... Yeah, but what does it identify? :p